src/control/XournalMain.cpp

```diff
@@ -346,7 +346,12 @@ auto XournalMain::run(int argc, char* argv[]) -> int {
         g_object_unref(file);
 
         if (!p.isEmpty()) {
-            opened = control->openFile(p, openAtPageNumber);
+            if (g_file_test(optFilename[0], G_FILE_TEST_EXISTS)) {
+                opened = control->openFile(p, openAtPageNumber);
+            } else {
+                GString* fileName = g_string_new(optFilename[0]);
+                opened = control->newFile("", fileName->str);
```

Comment:
Don't create a GString on the heap, this is a memory leak. Just pass optFilename.
```suggestion
                opened = control->newFile("", optFilename);
```

Reply:
Good point. My intention was to make it more readable, but that's not really worth it in this case.